src/main/kotlin/com/makarytskyi/rentcar/service/CarService.kt

```diff
@@ -0,0 +1,35 @@
+package com.makarytskyi.rentcar.service
+
+import com.makarytskyi.rentcar.dto.car.CarResponse
+import com.makarytskyi.rentcar.dto.car.CreateCarRequest
+import com.makarytskyi.rentcar.dto.car.UpdateCarRequest
+import com.makarytskyi.rentcar.exception.CarNotFoundException
+import com.makarytskyi.rentcar.repository.impl.SimpleCarRepository
+import org.springframework.stereotype.Service
+
+@Service
+class CarService(private val repository: SimpleCarRepository) {
+
+    fun findById(id: String): CarResponse = repository.findById(id)?.toResponse()
+        ?: throw CarNotFoundException("Car with id $id is not found")
+
+    fun findAll(): List<CarResponse> = repository.findAll().map { it.toResponse() }
+
+    fun save(car: CreateCarRequest): CarResponse = repository.findByPlate(car.plate)
+        ?.let { throw IllegalArgumentException("Car with plate ${car.plate} is already exist") }
+        ?: repository.save(car.toEntity()).toResponse()
```

Comment:
usually 'save' method means 'create or update'. if you don't allow updating here - please rename the method accordingly to not confuse other developers